ios/FluentUI/Vnext/List/List.swift

```diff
@@ -97,41 +159,102 @@ public struct MSFListView: View {
     }
 }
 
-/// UIKit wrapper that exposes the SwiftUI List implementation
-@objc open class MSFList: NSObject, FluentUIWindowProvider {
+/// Properties that make up section content
+class MSFListSectionStateImpl: NSObject, ObservableObject, Identifiable, MSFListSectionState {
+    func createCell() -> MSFListCellState {
+        return createCell(at: cells.endIndex)
+    }
+
+    func createCell(at index: Int) -> MSFListCellState {
+        guard index <= cells.count && index >= 0 else {
+            preconditionFailure("Index is out of bounds")
+        }
+        let cell = MSFListCellState()
+        cells.insert(cell, at: index)
+        return cell
+    }
+
+    func getCellState(at index: Int) -> MSFListCellState {
+        guard cells.indices.contains(index) else {
+            preconditionFailure("Index is out of bounds")
+        }
+        return cells[index]
+    }
 
-    @objc public init(sections: [MSFListSectionState],
-                      theme: FluentUIStyle? = nil) {
+    func removeCell(at index: Int) {
+        guard cells.indices.contains(index) else {
+            preconditionFailure("Index is out of bounds")
+        }
+        cells.remove(at: index)
+    }
+
+    var id = UUID()
+    @Published var cells: [MSFListCellState] = []
```

Comment:
Opinion: since this isn't meant to be editable outside this class, how would you feel about a `private(set)`? So this line could become

```swift
@Published private(set) var cells: [MSFListCellState] = []
```

That would keep us from accidentally writing code later on that directly modifies this array.